src/ui/preferences-window/LabelAndControl.swift

```diff
@@ -260,12 +260,14 @@ class LabelAndControl: NSObject {
                 button.setWidth(segmentWidth, forSegment: i)
             }
             if #available(macOS 11.0, *) {
-                if let preference = preference as? SfSymbolMacroPreference {
-                    button.setImage(NSImage(systemSymbolName: preference.symbolName, accessibilityDescription: nil)!, forSegment: i)
+                if let preference = preference as? SfSymbolMacroPreference,
+                   let symbolImage = NSImage(systemSymbolName: preference.symbolName, accessibilityDescription: nil) {
+                    button.setImage(symbolImage, forSegment: i)
                 }
             }
-            if defaults.int(rawName) == i {
-                button.selectedSegment = i
+            let selectedSegment = defaults.int(rawName)
+            if selectedSegment >= 0 && selectedSegment < macroPreferences.count {
+                button.selectedSegment = selectedSegment
```

Comment:
I'm not sure this helps. `i` is generated by the loop and its `enumerated()`. It's thus garanteed to be `> 0` and `< macroPreferences.count`.

Furthermore, the previous code was only assigning a value if it matches. I don't think it could assign a wrong value and create issues here.

Finally, the crash reports mention line `AppearanceTab.swift:414` which creates the UI for `appearanceSize`. However, in those reports, we see that users don't have this preference set. So it doesn't seem like this would be the root cause.

I've invited you on AppCenter. Let me know if it works 👍

Reply:
let selectedSegment = defaults.int(rawName)

The value stored by the user may have problems. The main reason for core is not here, but this code will become more robust

Reply:
`defaults.int(rawName) == i` protects us from bad data. It garantees that if we assign, it's a valid value for us.

I agree that we should have a better way to handle preferences. The worst releases we've had over the years have always been about bad preferences migration. In those cases, the app can't launch. It's the worse case.

I've added bf857e8f251cce38cb38d063161bf6a90c4a14a1 at some point. It makes it so a preferences that's of the wrong type will be removed, so the user isn't stuck. However it's only a type-check. We don't check that that value is valid beyond the type. It would be nice to add more safety in a generic way instead of checks all over the codebase when the preference is used. I'm not sure how to go about it though

Reply:
> I've added [bf857e8](https://github.com/lwouis/alt-tab-macos/commit/bf857e8f251cce38cb38d063161bf6a90c4a14a1) at some point. It makes it so a preferences that's of the wrong type will be removed, so the user isn't stuck. However it's only a type-check. We don't check that that value is valid beyond the type. It would be nice to add more safety in a generic way instead of checks all over the codebase when the preference is used. I'm not sure how to go about it though

A universal validation method is a good idea. There are many such frameworks in other languages, but ours will not be too complicated. We don't need to introduce additional frameworks. We just need to roughly implement a version.